### DISCUSSION TITLE
Installing on Debian 12

### DISCUSSION THREAD

**Discussion by rwb196884** [Netatalk/netatalk#836] [Q&A]
Tried to follow the inctructions at https://netatalk.io/docs/Installing-Netatalk-2-on-Debian

The first time it worked and `systemctl status netatalk` says `active` but nothing can connect.

After trying https://gist.github.com/agrlh/7618ed3e46fb02388022 `make all` now fails.

```
# pwd
/root/netatalk
# git status
On branch branch-netatalk-2-3
nothing to commit, working tree clean
# ./configure
...
# make all
make  all-recursive
make[1]: Entering directory '/root/netatalk'
Making all in include
make[2]: Entering directory '/root/netatalk/include'
Making all in atalk
make[3]: Entering directory '/root/netatalk/include/atalk'
make  all-am
make[4]: Entering directory '/root/netatalk/include/atalk'
make[4]: Nothing to be done for 'all-am'.
make[4]: Leaving directory '/root/netatalk/include/atalk'
make[3]: Leaving directory '/root/netatalk/include/atalk'
make[3]: Entering directory '/root/netatalk/include'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/root/netatalk/include'
make[2]: Leaving directory '/root/netatalk/include'
Making all in libatalk
make[2]: Entering directory '/root/netatalk/libatalk'
Making all in acl
make[3]: Entering directory '/root/netatalk/libatalk/acl'
  CC       libacl_la-cache.lo
  CC       libacl_la-unix.lo
  CC       libacl_la-uuid.lo
  CC       libacl_la-ldap.lo
  CC       libacl_la-ldap_config.lo
  CCLD     libacl.la
ar: /root/netatalk/libatalk/acl/.libs/libacl.a: No such file or directory
make[3]: *** [Makefile:482: libacl.la] Error 9
make[3]: Leaving directory '/root/netatalk/libatalk/acl'
make[2]: *** [Makefile:633: all-recursive] Error 1
make[2]: Leaving directory '/root/netatalk/libatalk'
make[1]: *** [Makefile:524: all-recursive] Error 1
make[1]: Leaving directory '/root/netatalk'
make: *** [Makefile:456: all] Error 2
```

**Comment by rdmark** [comment]
To see the status of netatalk v2 you need to run `systemctl status afpd` … because the “netatalk” service is what netatalk v3 uses but not v2. 

To assist with the connection issue please tell us:

- the contents of your afpd.conf
- Which Mac OS versions you’re connecting from
- And in the case of classic Mac OS, also what version of the AppleShare client you have installed. 

The most likely cause is that you don’t have the right UAM active for your clients. 

See also:
- https://netatalk.io/docs/Connect-to-AFP-Server-from-Mac-clients
- https://netatalk.io/oldstable/htmldocs/configuration#authentication

Now the make error is easy: just run `make clean` and start over.

**Comment by rdmark** [comment]
@dgsga I assume you’ve also seen the libacl.a issue in the past? I’m hoping it will just go away when moving to the Meson build system… but I’m considering trying to fix the automake bug anyways since it’s so obnoxious.

**Comment by ghost** [comment]
Hmm, I can't say I have. have been testing 2.3.1 on Deb 12.5 and it compiles fine. As for the issue going away with meson the answer is a resounding yes.

**Comment by rdmark** [comment]
@dgsga FWIW I’ve observed this on v3 branches too. It’s not solely a v2 phenomenon.

**Comment by rdmark** [comment]
Reported 9 years ago on the mailing lists: https://sourceforge.net/p/netatalk/mailman/message/32583063/

I agree with Ralph that it looks like an automake bug.

@rwb196884 Are you by any chance doing this on an RPi or other ARM based device?

**Comment by rwb196884** [comment]
Debian 12 on an i5.

I have a `netatalk` service -- to which nothing can connect -- and no `afpd`.

Is this right?
```
# git branch
* branch-netatalk-2-3
  main
```

And I guess I'll need to remove `netatalk`?

**Comment by rdmark** [comment]
Just to confirm: do you have a particular reason for running netatalk v2 over netatalk v3?

**Comment by rwb196884** [comment]
My main Macs are an SE/30 and an LCIII. It's nicer if they can see the server without having to type the IP address. 
Although getting either to work on Debian 12 would be better than nothing.

**Comment by rdmark** [comment]
Fair enough, then 2.x is the better choice.

First, you need to stop the existing netatalk process, and uninstall it cleanly. Do you remember how you installed it? Via `apt`?

Try stopping the process with:

```
sudo systemctl stop netatalk
```

Confirm it is stopped:

```
systemctl status netatalk
```

Attempt to uninstall with apt:

```
sudo apt remove netatalk
```

**Comment by rwb196884** [comment]
I don't remember how I installed it -- netatalk was removed from Debian 12.
Will
```
# systemctl stop netatalk
# systemctl disable netatalk
```
suffice?
(It's at `/usr/local/sbin/netatalk` and the version is `3.1.99dev`.)

**Comment by rdmark** [comment]
This means that you compiled it from source. If you still have the netatalk source code that you used, you can run `sudo make uninstall` from the root of the source code directory. 

It’s cleaner if you uninstall one version before installing another one.

**Comment by rwb196884** [comment]
Switched branch to `main` and run `make uninstall` has removed `/usr/local/sbin/netatalk` but the `systemd` service is still there and running -- after `systemctl daemon-reload`. And it even stops and restarts!

Seems to have left a symlink in `/etc/systemd/system/multi-user.target.wants` to a file that doesn't exist
```
netatalk.service -> /lib/systemd/system/netatalk.service
```
but even after removing it `systemctl start netatalk` works and shows as `active (exited)`.

I found `/etc/init.d/netatalk`
which contains
```
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
DESC="Netatalk"
NAME=netatalk
SCRIPTNAME=/etc/init.d/$NAME
DAEMON=/usr/sbin/netatalk
PIDFILE=/var/run/lock/netatalk
```
but `/usr/sbin/netatalk` doesn't exist! But `systemctl start netatalk` works!
I can't work out why `systemd` thinks that `netatalk` still exists and goodness knows what `systemctl start netatalk` is doing!

**Comment by rdmark** [comment]
`/etc/init.d/netatalk` is a traditional sysv style init script, and nothing to do with systemd. It’s odd that you have both on your system. There seems to be residue from multiple installations of netatalk with different prefixes and configurations...

`systemctl disable netatalk` should clean up all the symlinks and crap that systemd manages. There is no need to manually remove those. Doing so will only confuse systemd. 

You can do a global search on the file system for “netatalk” to see if there are other binaries floating around. 

And, use `ps aux` to inspect what processes are actually running. Do you see netatalk running after a reboot, for instance?

**Comment by rdmark** [comment]
Let me convert this to a Discussion since it a troubleshooting thread rather than a bug.

**Comment by rwb196884** [comment]
Trying again...

Switched to branch `branch-netatalk-2-4` and `apt install mesa` and followed the instructions in `INSTALL.md`.

Service is now running, but still nothing can connect (System 7.6 , MacOSX 10.7 connecting to `afp://...`).
Nothing in `/var/log/syslog`.

**Reply by rdmark** [reply]
My apologies, I forgot to respond to this!

Were you able to get it working in the end? What you are describing are usual caused by one of two reasons: Either you didn’t define any shared volutes, or the Netatalk server isn’t providing an authentication method that the Mac client understands. 

The next step would be to increase the log level and grab some logs. Please let me know if you want pointers.
